Core/Src/ws2812.cpp

```diff
@@ -73,32 +73,13 @@ void write_color_bytes_to_buffer(uint8_t index, uint8_t green_byte, uint8_t red_
 	}
 }
 
-//Writes the color specified by "color_code" to the neopixel specified by the index.
-void set_led_color_by_index(uint8_t index, uint32_t color_code) {
-
-	//Get the brightness of each color
-	uint8_t green_brightness_byte = get_green_byte_from_hex(color_code);
-	uint8_t red_brightness_byte = get_red_byte_from_hex(color_code);
-	uint8_t blue_brightness_byte = get_blue_byte_from_hex(color_code);
-
-	write_color_bytes_to_buffer(index, green_brightness_byte, red_brightness_byte, blue_brightness_byte, color_code);
-}
-
-
-//set the color for all 6 neopixels
-void set_all_led_colors(uint32_t color_code) {
-	for (int i = 0; i < NUM_LEDS; i++) {
-		set_led_color_by_index(i, color_code);
+void WS2812::set_all_led_colors(uint8_t green_byte, uint8_t red_byte, uint8_t blue_byte) {
+	for (int led = 0; led < NUM_LEDS; led++) {
+		set_led_color_by_index(led, green_byte, red_byte, blue_byte);
 	}
 }
 
-//Display the colors.
-void render_led_colors() {
-	HAL_TIMEx_PWMN_Start_DMA(&htim1, TIM_CHANNEL_2, (uint32_t*) out_buf,
-					BUFF_SIZE);
-}
-
-//TODO: Global flag for if DMA transfer has finished?
-void HAL_TIM_PWM_PulseFinishedCallback(TIM_HandleTypeDef *htim) {
 
+void WS2812::render_leds() {
+	HAL_TIMEx_PWMN_Start_DMA(&htim1, TIM_CHANNEL_2, (uint32_t*) out_buf, BUFF_SIZE);
```

Comment:
A good addition would be adding fields to the WS2812 class init that accepts a TIM_HandleTypeDef, TIM_CHANNEL, and pointers to the buffers.

Here it doesn't really  matter because the hardware is always consistent, but it would be good programming.

You can take a look at this: https://github.com/UWARG/efs-zeropilot-3.5/pull/63/files#:~:text=ICM42688(SPI_HandleTypeDef%20*%20SPI_HANDLE%2C%20GPIO_TypeDef%20*%20CS_GPIO_PORT%2C%20uint16_t%20CS_PIN)%3B